### PR TITLE
Mongocreateactivity

### PR DESCRIPTION
**Pull request by balajivemo**
I have checked in artifacts for a new activity called InsertDocument. 
I have also fixed a couple of minor bugs for other existing activities:
  - Name change of MongoDB Trigger
 - Changed datatype of timeout field from int to number in descriptor.json

@skothari-tibco and @fm-tibco Could you both review and provide your comments? Thanks!

### CODE REVIEW

File: mongodb/activity/insertDocument/metadata.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
+package insertDocument
+
+import (
+	"github.com/project-flogo/core/data/coerce"
+)
+
+// Settings structure
+type Settings struct {
+	Connection     string `md:"connection,required"`     // The MongoDB connection
+	Operation      string `md:"operation,required"`      // Operation to perform: Insert One Document, Insert Many Documents
```
Comment (skothari-tibco) [inline mongodb/activity/insertDocument/metadata.go]: May be add allowed in `md` tag too.

File: mongodb/activity/insertDocument/activity.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+				/* TODO: Handle this scenario better
+				Since API does not properly return list of InsertedIDs in case of continue on Error = false
+				and there was an error in one of the inserts, there is no good way to handle this case.
+				Users will see incorrect data in the output.
+				*/
+				logInsert.Warn(
+					"Error encountered during Insert Many Documents with Continue on Error set to false."+
+						"Users should check the mongoDB backend for the result of this particular execution for clarity."+
+						"Original error for reference : [%s] ", msg)
+			}
```
Comment (skothari-tibco) [inline mongodb/activity/insertDocument/activity.go:173]: You can create a function to handle this.

File: mongodb/activity/insertDocument/metadata.go (hunk 1 above)
Comment (balajivemo) [reply mongodb/activity/insertDocument/metadata.go]: Yes, will do

File: mongodb/activity/insertDocument/activity.go (hunk 2 above)
Comment (balajivemo) [reply mongodb/activity/insertDocument/activity.go:173]: @skothari-tibco I would like to have this piece of code within the eval itself. The number of lines of code is not much and it's a bit exaggerated by the lengthy log messages I am printing out for each case. I feel it's better to keep it within the eval

File: mongodb/activity/insertDocument/metadata.go (hunk 1 above)
Comment (balajivemo) [reply mongodb/activity/insertDocument/metadata.go]: Have done this for metadata.go of all activities. Pushed my changes in.

**Comment by balajivemo** [comment]
I had to push a few bug fixes raised by FE MongoDB QA as part of the PR. Can you check and provide your comments?

File: mongodb/trigger/mongodb/trigger.go [hunk 3]
```diff
@@ -204,11 +204,9 @@ func (evntLsnr *EventListener) listen() {
 			} else {
 				err := evntLsnr.stream.Err()
 				if err != nil {
-					//if err is not nil, it means something bad happened, let's finish our func
-					evntLsnr.logger.Errorf("Error while listening to the MongoDB event stream %s", err)
-					evntLsnr.stream.Close(context.Background())
-					evntLsnr.logger.Infof("Stopped Listening...stream closed")
-					return
+					//if err is not nil, don't stop listening on the event stream in case the connection comes back up.
+					evntLsnr.logger.Errorf("Error while listening to the MongoDB event stream . Detailed Error: %s", err)
```
Comment (skothari-tibco) [inline mongodb/trigger/mongodb/trigger.go]: If you are removing the closing of stream here, shouldn't you add it somewhere? Trigger Sop would be ideal.

Comment (balajivemo) [reply mongodb/trigger/mongodb/trigger.go]: oh yes, you are right.. Will add it to stop() method

Comment (fm-tibco) [reply mongodb/trigger/mongodb/trigger.go]: In addition to this shouldn't you send a "true" to the done channel before closing the stream?

**Review by fm-tibco — APPROVED** [review]

**Pull request merged**